test/metabase/pulse/render/body_test.clj

```diff
@@ -466,31 +466,6 @@
             {:cols default-multi-columns
              :rows [[10.0 1 1231 1] [5.0 10 nil 111] [2.50 20 11 1] [1.25 nil 1231 11]]})))))
 
-(deftest series-with-color-test
```

Comment:
These test aren't valid anymore, since now colors are determined on FE using either one or the combination of
- `seriesKey`
- `column`
- `series_settings`

Reply:
> Hmm... this might be true, but perhaps we should re-write tests to check that the js actually uses color data sent in via any of those keys.

Do you mean writing FE tests for that? Since I'm not sure how that could be done in the BE. This is like changing the contract between FE and BE, and when BE changes, FE needs to change too.

Reply:
> I think we can approve without it being covered here, and I can always add a small todo (I have a few tests I wanted to clean up in the time between pods anyway, and can take this one)

Thanks Adam!

Reply:
@WiNloSt My idea for a unit test on the backend is basically these steps:

1. create some card data with custom colors in the viz-settings
2. run the test renderer, which runs the static viz js, but instead of rendering out, it grabs the html/svg markup and parses it as a 'hiccup' tree (a popular tree data structure in Clojure land)
3. Grab the appropriate nodes (eg. a 'rect' node if it's a bar chart, 'path' for line, etc.) and read its props/attrs. Get :fill and see that it has the appropriate color attached

So, it's just a unit test that asserts 'Yes, the custom colors we've sent to the JS is, indeed used in the render"

Reply:
@adam-james-v That's very clear. Thanks for the explanation. I agree those tests would be good to have.